packages/astro/src/types/public/integrations.ts

```diff
@@ -248,10 +245,41 @@ export interface AstroIntegration {
  */
 export type IntegrationRouteData = Omit<
 	RouteData,
-	'isIndex' | 'fallbackRoutes' | 'redirectRoute'
+	'isIndex' | 'fallbackRoutes' | 'redirectRoute' | 'origin'
 > & {
 	/**
 	 * {@link RouteData.redirectRoute}
 	 */
 	redirectRoute?: IntegrationRouteData;
 };
+
+export interface IntegrationResolvedRoute {
+	/**
+	 * The current **pattern** of the route. For example:
+	 * - `src/pages/index.astro` has a pattern of `/`
+	 * - `src/pages/blog/[...slug].astro` has a pattern of `/blog/[...slug]`
+	 * - `src/pages/site/[blog]/[...slug].astro` has a pattern of `/site/[blog]/[...slug]`
+	 */
+	pattern: RouteData['route'];
+
+	/**
+	 *  Source component URL
+	 */
+	entrypoint: RouteData['component'];
```

Comment:
Why do we change the internal naming? Wouldn't it create fragmentation? Also, we shifted into a pattern where "entrypoint" is `URL` or `string`, where the `string` is an absolute file path. In this case, these are `string`, and they are relative paths. I think we should keep `component`, so they are not misunderstood like the other "entrypoints" we have in other APIs.

Reply:
I proposed changing the "component" name for the new public API because the value here is not necessarily a component. This is already true of `RouteData['component']`, the name is misleading.

I'm not sure whether `entrypoint` is the best name either. It can be:
- A relative path when the route is a redirect
- The absolute path to an Astro component when the route is a page
- The import path (before resolving dependencies) of an Astro component when the route is a page
- The absolute path to a JS file when the route is an API
- The import path (before resolving dependencies) of a JS file when the route is an API

It is quite an overloaded field, so I think it needs a more generic name than "component".

Some options and my thoughts about them:
- `source`: would be misleading for redirect since it is where it redirects _to_ and not _from_
- `definition`: I don't think it really fits with redirect
- `implementation`: also doesn't fit with redirect
- `target`: seems weird to say an Astro component is the target of a page but could work
- `reference`: in the sense of a "reference implementation" for the page or API and the reference link for the redirect

Reply:
What about `origin` or `creator`?

Reply:
cc @Fryuni

Reply:
Similarly, they feel weird for redirects because it is where you'd redirect _to_ not _from_. Both would imply _from_ to me.

I don't have any particularly strong feelings against any of the names Ema and I proposed, nor with `component`. I don't think this is worth blocking progress if it comes to it. It is just that none of them matches all 3 cases that can exist for this field. `component` is just worse than the alternatives for me since it directly implies one of the options when there are another 2.

Reply:
Yeah we can agree there isn't a perfect name, so we need to be good at documenting :) 

I also don't have particularly strong feelings about a particular name